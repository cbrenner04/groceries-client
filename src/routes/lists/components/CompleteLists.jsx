import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import AcceptedLists from './AcceptedLists';
import TitlePopover from '../../../components/TitlePopover';

function CompleteLists(props) {
  const message = props.fullList ? (
    'Previously refreshed lists are marked with an asterisk (*).'
  ) : (
    <>
      These are the completed lists most recently created.&nbsp;
      <Link to="/completed_lists">See all completed lists here.</Link>&nbsp; Previously refreshed lists are marked with
      an asterisk (*).
    </>
  );
  return <AcceptedLists title={<TitlePopover title="Completed" message={message} />} completed={true} {...props} />;
}

CompleteLists.propTypes = {
  fullList: PropTypes.bool,
};

CompleteLists.defaultProps = {
  fullList: false,
};

export default CompleteLists;
