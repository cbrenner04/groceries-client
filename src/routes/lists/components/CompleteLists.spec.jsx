import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import CompleteLists from './CompleteLists';

describe('CompleteLists', () => {
  let props;

  const renderCompleteLists = (p) => {
    const history = createMemoryHistory();
    return render(
      <Router history={history}>
        <CompleteLists {...p} />
      </Router>,
    );
  };

  beforeEach(() => {
    props = {
      history: {
        push: jest.fn(),
      },
      userId: 1,
      completedLists: [
        {
          id: 1,
          name: 'foo',
          type: 'GroceryList',
          owner_id: 1,
          created_at: new Date('05/27/2020').toISOString(),
          completed: true,
          users_list_id: 1,
          refreshed: false,
        },
      ],
      setCompletedLists: jest.fn(),
      currentUserPermissions: {},
      setCurrentUserPermissions: jest.fn(),
    };
  });

  it('renders link to full list in blurb when fullList is false', async () => {
    props.fullList = false;
    const { container, getAllByRole, getByTestId } = renderCompleteLists(props);

    fireEvent.click(getByTestId('Completed-popover'));
    await waitFor(() => getByTestId('popover-content'));

    expect(container).toMatchSnapshot();
    const allLinks = getAllByRole('link');
    expect(allLinks[allLinks.length - 1]).toHaveTextContent('See all completed lists here');
  });

  it('does not render link to full list when fullList is true', async () => {
    props.fullList = true;
    const { container, getAllByRole, getByTestId } = renderCompleteLists(props);

    fireEvent.click(getByTestId('Completed-popover'));
    await waitFor(() => getByTestId('popover-content'));

    expect(container).toMatchSnapshot();
    const allLinks = getAllByRole('link');
    expect(allLinks[allLinks.length - 1]).not.toHaveTextContent('See all completed lists here');
  });
});
