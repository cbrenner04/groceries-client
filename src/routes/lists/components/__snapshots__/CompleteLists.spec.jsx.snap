// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompleteLists does not render link to full list when fullList is true 1`] = `
<div>
  <div
    class="mb-4"
  >
    <div
      class="clearfix"
    >
      <div
        class="mx-auto float-left"
      >
        <div
          class="row"
          style="margin: 0px;"
        >
          <h2
            class="text-capitalize"
          >
            Completed
          </h2>
          <i
            class="far fa-question-circle float-right text-secondary"
            data-test-id="Completed-popover"
          />
        </div>
      </div>
      <button
        class="mx-auto float-right btn btn-link"
        type="button"
      >
        
         Select
      </button>
    </div>
    <div
      class="list-group"
    >
      <div
        class="accepted-list list-group-item"
        data-test-class="completed-list"
        data-test-id="list-1"
        style="display: block;"
      >
        <div
          class="row"
        >
          <div
            class="col-sm-12 col-12"
          >
            <div
              class="row"
            >
              <div
                class="pt-1 col-md-6"
              >
                <a
                  class="router-link"
                  href="/lists/1"
                >
                  <h5
                    class="mb-1"
                  >
                    <i
                      class="fa fa-shopping-bag text-info mr-3"
                    />
                    foo
                  </h5>
                </a>
              </div>
              <div
                class="pt-1 col-md-4"
              >
                <small
                  class="text-muted"
                >
                  May 27 2020, 12:00:00 am
                </small>
              </div>
              <div
                class="col-md-2"
              >
                <div
                  class="float-right btn-group"
                  role="group"
                >
                  <button
                    class="p-0 mr-3 btn btn-link"
                    data-test-id="complete-list-refresh"
                    style="pointer-events: auto; opacity: 1;"
                    type="button"
                  >
                    <i
                      class="fa fa-redo fa-2x text-primary"
                    />
                  </button>
                  <button
                    class="p-0 btn btn-link"
                    data-test-id="complete-list-trash"
                    type="button"
                  >
                    <i
                      class="fa fa-trash fa-2x text-danger"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CompleteLists renders link to full list in blurb when fullList is false 1`] = `
<div>
  <div
    class="mb-4"
  >
    <div
      class="clearfix"
    >
      <div
        class="mx-auto float-left"
      >
        <div
          class="row"
          style="margin: 0px;"
        >
          <h2
            class="text-capitalize"
          >
            Completed
          </h2>
          <i
            class="far fa-question-circle float-right text-secondary"
            data-test-id="Completed-popover"
          />
        </div>
      </div>
      <button
        class="mx-auto float-right btn btn-link"
        type="button"
      >
        
         Select
      </button>
    </div>
    <div
      class="list-group"
    >
      <div
        class="accepted-list list-group-item"
        data-test-class="completed-list"
        data-test-id="list-1"
        style="display: block;"
      >
        <div
          class="row"
        >
          <div
            class="col-sm-12 col-12"
          >
            <div
              class="row"
            >
              <div
                class="pt-1 col-md-6"
              >
                <a
                  class="router-link"
                  href="/lists/1"
                >
                  <h5
                    class="mb-1"
                  >
                    <i
                      class="fa fa-shopping-bag text-info mr-3"
                    />
                    foo
                  </h5>
                </a>
              </div>
              <div
                class="pt-1 col-md-4"
              >
                <small
                  class="text-muted"
                >
                  May 27 2020, 12:00:00 am
                </small>
              </div>
              <div
                class="col-md-2"
              >
                <div
                  class="float-right btn-group"
                  role="group"
                >
                  <button
                    class="p-0 mr-3 btn btn-link"
                    data-test-id="complete-list-refresh"
                    style="pointer-events: auto; opacity: 1;"
                    type="button"
                  >
                    <i
                      class="fa fa-redo fa-2x text-primary"
                    />
                  </button>
                  <button
                    class="p-0 btn btn-link"
                    data-test-id="complete-list-trash"
                    type="button"
                  >
                    <i
                      class="fa fa-trash fa-2x text-danger"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
